### DISCUSSION TITLE
Copilot: Disabled for filetype=help by internal default

### DISCUSSION THREAD

**Discussion by MorganBergen** [community/community#38074] [Copilot Conversations] [answered]
`:Copilot version`

```CLI
copilot.vim 1.5.3
Neovim 0.8.0
copilot/dist/agent.js 1.43.0
Node.js 17.9.1
```
I would like to set markdown to be enabled by default, however I'm new to neovim and I dont know where exactly I should set that global dictionary mapping 

```
let g:copilot_filetypes = {
    \ 'xml': v:false,
    \ }
```

Where do I set this in the `copilot.txt` or in some options configuration?
I just dont know what command to run in order to set all files for copilot to be true

```
let g:copilot_filetypes = {
    \ '*' : v:true,
    \ }
```

<img width="1440" alt="Screenshot 2022-11-02 at 9 21 03 AM" src="https://user-images.githubusercontent.com/65584733/199524112-88ace0d2-55fd-4fcd-8275-81ca5a751815.png">

**Comment by MorganBergen** [comment]
I'm not sure if I write out this global mapping inside the options

**Comment by maxpatiiuk** [comment]
I believe the configuration should be placed into `~/.config/nvim/init.vim` (create that file if it doesn't already exist) ([example nvim configuration](https://dev.to/dafloresdiaz/neovim-for-macos-3nk0))

Also, on your 2nd screenshot it says `most file types are enabled by default`, thus I don't know if you have to manually enable copilot at all.

**Comment by tpope** [comment, marked as answer]
It goes in your vimrc, generally `~/.config/nvim/init.vim` as @maxxxxxdlp notes.  Try this:

```vim
let g:copilot_filetypes = {'markdown': v:true}
```

If you're using a pure Lua config like `~/.config/nvim/init.lua`, you can use a Lua variant:

```lua
vim.g.copilot_filetypes = {markdown = true}
```

**Reply by toomey8** [reply]
aha! This works. Thanks!

**Reply by benlieb** [reply]
This works! Could we add this to the README? I Can make a PR if you want. Working examples are so helpful!

**Reply by Pclucido** [reply]
Sure

**Reply by anandkumar89** [reply]
how to disable for all filetypes in lua ?

**Reply by 0guanhua0** [reply]
[filetype_defaults](https://github.com/github/copilot.vim/blob/87038123804796ca7af20d1b71c3428d858a9124/autoload/copilot.vim#L124)

**Comment by debragail** [comment]
doesn't work for me unfortunately still get the same error

**Reply by Pclucido** [reply]
Is it a 404 or 552 error

**Reply by debragail** [reply]
I can't remember now.  I can't confirm or deny.

**Reply by richardxgao** [reply]
This works for me as per `:help copilot`

```lua
require("copilot").setup {
  filetypes = {
    markdown = true, -- overrides default
    terraform = false, -- disallow specific filetype
    sh = function ()
      if string.match(vim.fs.basename(vim.api.nvim_buf_get_name(0)), '^%.env.*') then
        -- disable for .env files
        return false
      end
      return true
    end,
  },
}
```

**Comment by tpope** [comment]
The people have spoken. The file types `yaml`, `markdown`, and `help` are now turned on by default!

**Comment by jakemayfield** [comment]
What if the configuration is needed on a per project basis? 

Does Copilot recognize these settings placed inside of a `.vimrc` or `init.vim` file placed in a project's root directory?

**Comment by Ding-Fan** [comment]
If you are using astronvim with [copilot-cmp](https://github.com/AstroNvim/astrocommunity/blob/main/lua/astrocommunity/completion/copilot-cmp/README.md) , try create a `~/.config/nvim/lua/plugins/copilot-override.lua` with content like this

```lua
return {
  "zbirenbaum/copilot.lua",
  opts = {
    filetypes = {
      markdown = true, -- Explicitly enable markdown
    },
  },
}
```

**Comment by rlanore** [comment]
For other to search how to disable all file types in lua

```lua
vim.g.copilot_filetypes = {
    [ '*' ] = false,
    terraform = true
}
```
